### DISCUSSION TITLE
clangd in vscode does not offer suggestions in agregate intializer lists?

### DISCUSSION THREAD

**Discussion by MatejSakmary** [clangd/clangd#1419] [Q&A] [answered]
Hello,

I often use structs to bundle parameters whenever I'm passing to a function as follows:

```cpp
struct FooInfo
{
    int i;
    int j;
    int k;
};

auto foo(const FooInfo & info) -> void  {...}
```
which I then like to call like this:
```cpp
foo({
    .i = 3,  // I get no suggestions here 
    .j = 4,  
    ...
});
```
However when calling this function and using the specified syntax I get no suggestions. Is this not possible with clangd or am I perhaps missing some setting?

Thanks a lot for answers :)

**Comment by HighCommander4** [comment, marked as answer]
I assume you're looking for completion of the member names after the `.` in the designator.

Clangd does offer this if you write out the type name, e.g. `FooInfo{.^}`.

In the case where the type name is omitted, there's a bit of a chicken an egg problem: clangd doesn't know what type you're dealing with until the name `foo` (which could name a function with multiple overloads) is resolved to a particular declaration, but it can't resolve the name `foo` to a particular declaration until the type of the argument expression is known.

In the case where `foo` has just a single overload in scope, it would be possible for clangd to heuristically assume that you're going to use that overload, and offer completions based on that. This has not been implemented.
